packages/dx-scheduler-core/src/plugins/editing-state/helpers.ts

```diff
@@ -0,0 +1,239 @@
+import moment from 'moment';
+import { RRule, rrulestr, RRuleSet, Options } from 'rrule';
+import { PureComputed } from '@devexpress/dx-core';
+import {
+  AppointmentModel, PreCommitChanges, ChangeSet, Changes, MakeDateSequence,
+} from '../../types';
+import { RECURRENCE } from '../../constants';
+
+export const deleteCurrent: PureComputed<
+  [Partial<AppointmentModel>], ChangeSet
+> = (appointmentData) => {
+  const currentSequence: Date[] = makeDateSequence(appointmentData.rRule, appointmentData.exDate, {
+    dtstart: moment.utc(appointmentData.parentData.startDate).toDate(),
+  });
+
+  if (currentSequence.length === 1) {
+    return deleteAll(appointmentData);
+  }
+
+  const currentExDate = `${moment.utc(appointmentData.startDate as Date)
```

Comment:
The `deleteCurrent` and `EditCurrent` methods have common code which can be extract into the extra method.